Author: Alastair McKinstry <mckinstry@debian.org>
Description: Build both static and dynamic libraries for Debian;
 Add SONAME support for the dynamic libraries
Last-Updated: 2016-04-27
Forwarded: no

--- a/fortran/CMakeLists.txt
+++ b/fortran/CMakeLists.txt
@@ -47,6 +47,17 @@
     ecbuild_add_library( TARGET          eccodes_f90
                          SOURCES         grib_fortran.c grib_f90.f90 eccodes_f90.f90 grib_kinds.h
                          GENERATED       grib_f90.f90 eccodes_f90.f90
+			 CFLAGS          ${CFLAGS}
+			 VERSION         0
+                         PUBLIC_INCLUDES $<BUILD_INTERFACE:${CMAKE_Fortran_MODULE_DIRECTORY}>
+                                         $<INSTALL_INTERFACE:${INSTALL_INCLUDE_DIR}>
+                         PRIVATE_LIBS    eccodes )
+    ecbuild_add_library( TARGET          eccodes_f90_static
+			 TYPE            STATIC
+                         SOURCES         grib_fortran.c grib_f90.f90 eccodes_f90.f90 grib_kinds.h
+                         GENERATED       grib_f90.f90 eccodes_f90.f90
+			 CFLAGS          ${CFLAGS}
+			 VERSION         0
                          PUBLIC_INCLUDES $<BUILD_INTERFACE:${CMAKE_Fortran_MODULE_DIRECTORY}>
                                          $<INSTALL_INTERFACE:${INSTALL_INCLUDE_DIR}>
                          PRIVATE_LIBS    eccodes ${ECCODES_PTHREADS_LIBRARIES} )
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -433,6 +433,7 @@
                      SOURCES   ${CMAKE_CURRENT_BINARY_DIR}/grib_api_version.cc
                               # griby.cc gribl.cc
                               ${eccodes_src_files}
+	             VERSION  0
                      #PRIVATE_LIBS      ${ECCODES_EXTRA_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${CMATH_LIBRARIES}
                      PRIVATE_INCLUDES "${AEC_INCLUDE_DIRS}" "${PNG_INCLUDE_DIRS}"
                      PRIVATE_LIBS ${ECCODES_EXTRA_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${AEC_LIBRARIES} ${PNG_LIBRARIES}
@@ -447,3 +448,25 @@
                     PUBLIC_INCLUDES
                               $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/src>
                               $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src> )
+
+ecbuild_add_library( TARGET    eccodes_static
+  		      TYPE      STATIC
+                     VERSION   0
+                     SOURCES   ${CMAKE_CURRENT_BINARY_DIR}/grib_api_version.cc
+                              # griby.cc gribl.cc
+                              ${grib_api_srcs}
+                     #PRIVATE_LIBS      ${ECCODES_EXTRA_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${CMATH_LIBRARIES}
+                     PRIVATE_INCLUDES "${AEC_INCLUDE_DIRS}"
+                     PRIVATE_LIBS ${ECCODES_EXTRA_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${AEC_LIBRARIES} ${PNG_LIBRARIES}
+                     PUBLIC_LIBS  ${CMATH_LIBRARIES} ${THREADS_LIBRARIES}
+                     TEMPLATES ${grib_api_extra_srcs}
+                     INSTALL_HEADERS_LIST
+                              grib_api.h
+                              eccodes.h
+                              eccodes_windef.h
+                              ${CMAKE_CURRENT_BINARY_DIR}/eccodes_version.h
+                              ${PROJECT_BINARY_DIR}/eccodes_config.h
+                    PUBLIC_INCLUDES
+                              $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/src>
+                              $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src> )
+			    
